source/main/gfx/DustManager.cpp

```diff
@@ -315,7 +315,17 @@ void RoR::GfxScene::RemoveGfxActor(RoR::GfxActor* remove_me)
         if (*itor == remove_me)
         {
             m_all_gfx_actors.erase(itor);
-            return;
+            break;
+        }
+    }
+    itor = m_live_gfx_actors.begin();

```

Comment:
needs a comment `//TODO:  this is a temporary workaround for actor removal being done when it shouldn't (legacy logic).`